front/pages/api/w/[wId]/index.ts

```diff
@@ -83,16 +85,27 @@ async function handler(
         });
       }
 
-      if (req.body.name) {
+      if ("name" in body) {
         await w.update({
-          name: req.body.name,
+          name: body.name,
         });
-        owner.name = req.body.name as string;
+        owner.name = body.name;
       } else {
-        await w.update({
-          allowedDomain: req.body.allowedDomain,
-        });
-        owner.allowedDomain = req.body.allowedDomain as string | null;
+        const user = auth.user();
+        if (body.autoAddDomainUsers && user) {
+          const [, userEmailDomain] = user.email.split("@");
```

Comment:
Should not be possible but who knows... but this does not get the domain if there is 2 `@` in the user.email ?

Reply:
We only support signing up through OAuth flows today, meaning that the email we have here is coming from either Google and GitHub. 

As this part won't live here for long, we will move it to the Workspace creating, maybe I can enforce a stricter check on the email when working on this subsequent PR, WDYT?